### PR TITLE
Add base structure of model zoo server

### PR DESCRIPTION
**Pull request by typhoonzero**
Follow up https://github.com/sql-machine-learning/sqlflow/pull/2180

### CODE REVIEW

**Comment by wangkuiyi** [comment]
Please see my comments https://github.com/sql-machine-learning/sqlflow/pull/2180#pullrequestreview-406745754. I am afraid that we don't yet have a clear entity-relationship in our mind when we start writing the gRPC
interface.  It could be easier to make progress by starting from the database schema design.

File: pkg/modelzooserver/modelzooserver_test.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		log.Fatalf("failed to listen: %v", err)
+	}
+	grpcServer := grpc.NewServer()
+	pb.RegisterModelZooServerServer(grpcServer, &modelZooServer{})
+
+	grpcServer.Serve(lis)
+}
+
+func serverIsReady(addr string, timeout time.Duration) bool {
+	conn, err := net.DialTimeout("tcp", addr, timeout)
```
Comment (lhw362950217) [inline pkg/modelzooserver/modelzooserver_test.go:40]: Shall we put these kind of tool func to a dedicated package？I noticed we have so many copies of them.

Comment (typhoonzero) [reply pkg/modelzooserver/modelzooserver_test.go:40]: Thanks for reminding, I'll refactor this in another PR.

**Review by lhw362950217 — APPROVED** [review]
LGTM

**Pull request merged**